src/query/condition.rs

```diff
@@ -260,6 +260,38 @@ impl Condition {
     pub fn len(&self) -> usize {
         self.conditions.len()
     }
+
+    pub(crate) fn to_simple_expr(&self) -> SimpleExpr {
+        let expr = if self.conditions.is_empty() {
+            SimpleExpr::Constant(match self.condition_type {
+                ConditionType::Any => false.into(),
+                ConditionType::All => true.into(),
+            })
+        } else {
+            let mut inner_exprs = vec![];
+            for ce in &self.conditions {
+                inner_exprs.push(match ce {
+                    ConditionExpression::Condition(c) => c.to_simple_expr(),
+                    ConditionExpression::SimpleExpr(e) => e.clone(),
+                });
+            }
+            let mut inner_exprs_into_iter = inner_exprs.into_iter();
+            // Guaranteed to exist a next() by case distinction on conditions empty
```

Comment:
We could change the .empty check with a match on next

Reply:
Done! Thanks for the tip :)